### PR TITLE
Don't alias when the source and target are the same symbol

### LINKED ISSUE #1955: Fuzz recursive alias of class method causes Segmentation fault
#### Input

[→ View on sorbet.run](https://sorbet.run/#%23%20typed%3A%20true%0A%0Aclass%20BadAliasingClassMethod1%0A%20%20class%20%3C%3C%20self%0A%20%20%20%20def%20hello%0A%20%20%20%20%20%20'HELLO%20WORLD'%0A%20%20%20%20end%0A%0A%20%20%20%20alias_method%20%3Ahello%2C%20%3Ahello%0A%20%20end%0Aend%0A%0Adef%20main%0A%20%20BadAliasingClassMethod1.hello%0Aend)

```ruby
# typed: true

class BadAliasingClassMethod1
  class << self
    def hello
      'HELLO WORLD'
    end

    alias_method :hello, :hello
  end
end

def main
  BadAliasingClassMethod1.hello
end
```

#### Observed output

The output hung up.

```
Loading...
```

Developer Tools's console shows these erros.

```
sorbet-wasm.js:8 Uncaught 17118816

sorbet-wasm.js:8 Uncaught TypeError: functionPointers[index] is not a function
    at jsCall_vi (sorbet-wasm.js:8)
    at /wasm-function[11996]:0x206b64
    at /wasm-function[8566]:0x16789e
    at sorbet-wasm.js:8
    at Object.ccall (sorbet-wasm.js:8)
    at WebSocket.<anonymous> (client.ts:170)
    at target.js:63
    at Array.forEach (<anonymous>)
    at WebSocket.dispatchEvent (target.js:61)
    at Server.<anonymous> (websocket.js:148)

console-window.ts:11 Couldn't start client Sample Language Client

main.js:138 Uncaught (in promise) Error: Connection is disposed.
    at new ConnectionError (main.js:138)
    at throwIfClosedOrDisposed (main.js:616)
    at Object.sendRequest (main.js:715)
    at Object.shutdown (connection.ts:107)
    at client.js:2260

sorbet-wasm.js:8 Uncaught Finished up all reserved function pointers. Use a higher value for RESERVED_FUNCTION_POINTERS.
```

`srb tc` shows below

```
jiku@AG400XN-M2S5:~/resraku$ uname -a
Linux AG400XN-M2S5 4.4.0-18362-Microsoft #1-Microsoft Mon Mar 18 12:02:00 PST 2019 x86_64 x86_64 x86_64 GNU/Linux

jiku@AG400XN-M2S5:~/resraku$ bundle exec srb --version
Sorbet typechecker 0.4.4869 git 05b12362d97ec5d25f360181cb66fc1435af9cc3-dirty built on 2019-10-15 05:32:00 GMT debug_symbols=true

jiku@AG400XN-M2S5:~/resraku$ cat bad_class_method_aliasing.rb
# typed: true

class BadAliasingClassMethod1
  class << self
    def hello
      'HELLO WORLD'
    end

    alias_method :hello, :hello
    # alias_method :world, :hello
  end
end

jiku@AG400XN-M2S5:~/resraku$ bundle exec srb tc --no-config --error-white-list 1000 bad_class_method_aliasing.rb
Exception::raise(): Unexpected type alias: AliasType { symbol = <S <C <U BadAliasingClassMethod1>> $1><U hello> }

Backtrace:
  #3 0x897002 sorbet::definition_validator::variance::VarianceValidator::validatePolarity()
  #4 0x896e6e sorbet::definition_validator::variance::VarianceValidator::validateMethod()
  #5 0x89265e sorbet::ast::TreeMapper<>::mapIt()
  #6 0x893ec1 sorbet::ast::TreeMapper<>::mapIt()
  #7 0x893ec1 sorbet::ast::TreeMapper<>::mapIt()
  #8 0x892425 sorbet::ast::TreeMapper<>::mapIt()
  #9 0x893ec1 sorbet::ast::TreeMapper<>::mapIt()
  #10 0x88ffe9 sorbet::definition_validator::runOne()
  #11 0x6adb58 std::__1::__function::__func<>::operator()()
  #12 0xd12278 sorbet::WorkerPoolImpl::multiplexJob()
  #13 0x692369 sorbet::realmain::pipeline::typecheck()
  #14 0x49e2c6 sorbet::realmain::realmain()
  #15 0x49884b main
  #16 0x7f8327051b97 __libc_start_main
  #17 0x45602a _start

/home/jiku/resraku/vendor/bundle/ruby/2.6.0/gems/sorbet-0.4.4869/bin/srb: line 46:  4502 Segmentation fault      (core dumped) "${sorbet[0]}" "${args[@]}"
```

#### Expected behavior

I found a test below.

[test/testdata/cfg/fuzz_recursive_alias.rb](https://github.com/sorbet/sorbet/commit/24fdd52301a25d137814662f1e5d6727697144b7#diff-0e6bd40fb5f57447188285cd110bd70eR3) 

It raises `error-with-dupes: Too many alias expansions for symbol ::Object#foo, the alias is either too long or infinite.` and Segmentation fault does not occur.

```
jiku@AG400XN-M2S5:~/resraku$ cat bad_instance_method_aliasing.rb
# typed: true

class BadAliasingInstanceMethod1
  def hello
    'HELLO WORLD'
  end

  alias_method :hello, :hello
  # alias_method :world, :hello
end
jiku@AG400XN-M2S5:~/resraku$ bundle exec srb tc --no-config --error-white-list 1000 bad_instance_method_aliasing.rb
bad_instance_method_aliasing.rb:4: Too many alias expansions for symbol ::BadAliasingInstanceMethod1#hello, the alias is either too long or infinite. Next expansion would have been to ::BadAliasingInstanceMethod1#hello https://srb.help/1003
     4 |  def hello
          ^^^^^^^^^
Errors: 1
```

When type checking an class method should be treated the same way.

- - -

I caught this error when type checking [sporkmonger/addressable lib/addressable/uri.rb#L424](https://github.com/sporkmonger/addressable/blob/addressable-2.7.0/lib/addressable/uri.rb#L424).

### PR DESCRIPTION
**Pull request by elliottt**
Fixes #1955 

### Motivation

We shouldn't alias symbols to themselves.

### Test plan

See included automated tests.

### CODE REVIEW

**Comment by DarkDimius** [comment]
Can we add a test?

**Comment by elliottt** [comment]
> Can we add a test?

That's why this was marked [wip] :)

File: test/testdata/resolver/invalid_alias.rb.symbol-table-raw.exp [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+      argument <blk><block> @ Loc {file=test/testdata/resolver/invalid_alias.rb start=??? end=???}
+    method <S <C <U BadAliasingClassMethod1>> $1><U hello> (<blk>) @ Loc {file=test/testdata/resolver/invalid_alias.rb start=5:5 end=5:14}
+      argument <blk><block> @ Loc {file=test/testdata/resolver/invalid_alias.rb start=??? end=???}
+  class <S <S <C <U BadAliasingClassMethod1>> $1> $1>[<C <U <AttachedClass>>>] < <S <S <C <U Object>> $1> $1> () @ Loc {file=test/testdata/resolver/invalid_alias.rb start=4:3 end=4:8}
+    type-member(+) <S <S <C <U BadAliasingClassMethod1>> $1> $1><C <U <AttachedClass>>> -> LambdaParam(<S <S <C <U BadAliasingClassMethod1>> $1> $1><C <U <AttachedClass>>>, lower=T.noreturn, upper=AppliedType {
+  klass = <S <C <U BadAliasingClassMethod1>> $1>
+  targs = [
+    <C <U <AttachedClass>>> = BadAliasingClassMethod1
+  ]
+}) @ Loc {file=test/testdata/resolver/invalid_alias.rb start=4:3 end=4:8}
```
Comment (jez) [inline test/testdata/resolver/invalid_alias.rb.symbol-table-raw.exp]: I would love to not print multi-line types in this single-line symbol-table format... 🤔

Comment (jez) [reply test/testdata/resolver/invalid_alias.rb.symbol-table-raw.exp]: i should note: this is very non-blocking.

Comment (elliottt) [reply test/testdata/resolver/invalid_alias.rb.symbol-table-raw.exp]: #1960

**Pull request merged**